src/Microsoft.VisualStudio.Services.Agent/Constants.cs

```diff
@@ -228,5 +229,17 @@ public static class Task
                 public static readonly string DisplayName = "task.displayname";
             }
         }
+
+        public class Release
+        {
+            public class Path
+            {
+                public static readonly string ReleaseDirectoryPrefix = "r";
```

Comment:
Again, i am not introducing a new concept here. _work is and wil be the work folder for both release and build.
Changing from 'r/1' to 'r1' will not fix the problem that you are referring to.
There wont be any variable that will point to work/r1. so still there is hope for a new variable.

system default working directory will point to work/r1/a
agent.releaseDirectory will point to work/r1/a
system.artifactsDirectory will point to work/r1/a

If we agree with using r{int}, then it would be make more sense for build to start following b{int} for folder names for the sake of consistency. I understand adding a r/{int} is no consistency but we assumed migrating from work/1/ to work/b/1 for build would be easier.
Ideally, there should be one count generator shared across plugins. This way we would be completely consistent.